### PR TITLE
Dependecy fixes, and MDX2

### PR DESCRIPTION
**Pull request by Tigge**
Upgrade to MDX 2. There are a few changes mainly to how the code and pre
blocks work. We are now using `remark-mdx-code-meta` to read meta from
the MDX code blocks. This required a bit of a rewrite to that handling
in docs.

Also:
- Update yarn sdks
- Unpin dependencies and peer dependencies
- Adjust `react-transition-group` peer dependency to 4.4.0 as we are
  using `nodeRef` already.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "module": "esnext",
     "moduleResolution": "node",
     "resolveJsonModule": true,
-    "jsx": "react",
+    "jsx": "react-jsx",
```
[the remark this answers is not in the document]
Comment (anius) [reply tsconfig.json:12]: Will be resolved by #209

File: packages/core/package.json [hunk 2]
```diff
@@ -46,9 +46,9 @@
   },
   "peerDependencies": {
     "pepjs": "^0.5.2",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "react-transition-group": "^4.3.0",
+    "react": "^17.0.0",
```
[the remark this answers is not in the document]
Comment (Tigge) [reply packages/core/package.json]: Reverted

File: packages/docs/src/mdx/Base.tsx [hunk 3]
```diff
@@ -5,6 +5,9 @@ import {
   Divider,
   Link,
 } from 'practical-react-components-core'
+import React from 'react'
```
[the remark this answers is not in the document]
Comment (Tigge) [reply packages/docs/src/mdx/Base.tsx]: Very true, fixed

File: packages/docs/package.json [hunk 4]
```diff
@@ -11,18 +11,18 @@
...
+    "@mdx-js/react": "2.0.0",
     "core-js": "3.21.1",
     "formik": "2.2.9",
     "practical-react-components-core": "workspace:*",
     "practical-react-components-formik": "workspace:*",
     "practical-react-components-icons": "workspace:*",
     "prism-react-renderer": "1.3.1",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
+    "react": "^17.0.1",
```
Comment (anius) [inline packages/docs/package.json:22]: Use exact version

File: packages/docs/package.json [hunk 5]
```diff
@@ -11,18 +11,18 @@
...
+    "react-dom": "^17.0.1",
```
Comment (anius) [inline packages/docs/package.json:23]: Use exact version

File: packages/docs/package.json [hunk 6]
```diff
@@ -11,18 +11,18 @@
...
     "react-hooks-shareable": "1.48.0",
-    "react-is": "17.0.2",
+    "react-is": "^17.0.1",
```
Comment (anius) [inline packages/docs/package.json:25]: Use exact version

File: packages/docs/src/mdx/coreComponents/Button.mdx [hunk 7]
```diff
@@ -44,39 +44,39 @@ export const Wrapper = styled.div`
 
 A button with the primary color of the theme.
 
-```typescript type=live
+```typescript type="live" test="woo"
```
[the remark this answers is not in the document]
Comment (anius) [reply packages/docs/src/mdx/coreComponents/Button.mdx:47]: So why not remove?

File: packages/docs/webpack.config.ts [hunk 8]
```diff
@@ -114,6 +122,7 @@ export default async (env: Record<string, string> = {}) => {
     },
     resolve: {
       extensions: ['.tsx', '.ts', '.js'],
+      fallback: { url: false },
```
Comment (anius) [inline packages/docs/webpack.config.ts:125]: Add some comment why this is needed?

File: packages/ui-tests/package.json [hunk 9]
```diff
@@ -12,10 +12,10 @@
     "core-js": "3.21.1",
     "practical-react-components-core": "workspace:*",
     "practical-react-components-icons": "workspace:*",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
+    "react": "^17.0.1",
```
Comment (anius) [inline packages/ui-tests/package.json:15]: Use exact version

File: packages/ui-tests/package.json [hunk 10]
```diff
@@ -12,10 +12,10 @@
...
+    "react-dom": "^17.0.1",
```
Comment (anius) [inline packages/ui-tests/package.json:16]: Use exact version

File: packages/ui-tests/package.json [hunk 11]
```diff
@@ -12,10 +12,10 @@
...
     "react-hooks-shareable": "1.48.0",
-    "react-is": "17.0.2",
+    "react-is": "^17.0.1",
```
Comment (anius) [inline packages/ui-tests/package.json:18]: Use exact version

**Review by anius — CHANGES_REQUESTED** [review]

**Review by Therealnaxos — APPROVED** [review]
"mob-reviewed", Zeno approves changes.

**Comment by anius** [comment]
@Tigge I've moved your code to #220 as I couldn't update this PR.

**Pull request closed without merging**